core/src/main/java/com/datastax/oss/driver/api/core/ssl/ProgrammaticSslEngineFactory.java

```diff
@@ -69,6 +69,24 @@ public ProgrammaticSslEngineFactory(
   /**
    * Creates an instance with the given {@link SSLContext}, cipher suites and host name validation.
    *
+   * <p>Note: {@linkplain
+   * com.datastax.oss.driver.api.core.config.DefaultDriverOption#SSL_HOSTNAME_VALIDATION host name
+   * validation} is not compatible with unresolved addresses. If this feature is enabled, make sure
+   * that you only provide resolved contact points:
+   *
+   * <ul>
+   *   <li>If you provided your {@linkplain
+   *       com.datastax.oss.driver.api.core.config.DefaultDriverOption#CONTACT_POINTS contact
```

Comment:
It mean that when set `com.datastax.oss.driver.api.core.config.DefaultDriverOption#CONTACT_POINTS` we are createing unresolved addresses, right?

Reply:
It depends on `advanced.resolve-contact-points` see `ContactPoints`.